cetlib/LibraryManager.cc

```diff
@@ -154,35 +154,37 @@ cet::LibraryManager::libraryIsLoadable(std::string const& path) const
 void
 cet::LibraryManager::lib_loc_map_inserter(std::string const& path)
 {
-  lib_loc_map_[boost::filesystem::path(path).filename().native()] = path;
+  lib_loc_map_[std::filesystem::path(path).filename().native()] = path;
 }
 
 void
 cet::LibraryManager::spec_trans_map_inserter(
   lib_loc_map_t::value_type const& entry)
 {
   // First obtain short spec.
-  boost::regex const e{"([^_]+)_" + lib_type_ + dllExtPattern() + '$'};
-  boost::match_results<std::string::const_iterator> match_results;
-  if (boost::regex_search(entry.first, match_results, e)) {
+  std::string e_str =
+    "([^_]+)_" + lib_type_ + dllExtPattern() + '$';
+  std::regex const e{e_str};
+  std::match_results<std::string::const_iterator> match_results;
+  if (std::regex_search(entry.first, match_results, e)) {
     spec_trans_map_[match_results[1]].insert(entry.second);
   } else {
     throw exception("LogicError")
       << "Internal error in spec_trans_map_inserter for entry " << entry.first
-      << " with pattern " << e.str();
+      << " with pattern " << e_str;
   }
   // Next, convert library filename to full libspec.
   std::ostringstream lib_name;
   std::ostream_iterator<char, char> oi{lib_name};
-  boost::regex_replace(oi,
+  std::regex_replace(oi,
                        entry.first.begin(),
                        entry.first.end(),
-                       boost::regex{"(_+)"},
-                       std::string{"(?1/)"},
-                       boost::match_default | boost::format_all);
-  boost::regex const stripper{"^lib(.*)/" + lib_type_ + "\\..*$"};
+                       std::regex{"_+"},
+                       std::string{"/"}
+                       );
+  std::regex const stripper{"^lib(.*)/" + lib_type_ + "\\..*$"};
```

Comment:
I can't imagine a scenario where I would have willingly used recursive capture syntax (`(?1/)`) without having a case that required it: but if that were the case, I should have made a regression test for it. Removing it still leaves me uneasy, though.

Reply:
Yeah, I'm really confused about this bit, too...https://www.boost.org/doc/libs/1_83_0/libs/regex/doc/html/boost_regex/format/boost_format_syntax.html appears to imply that this is the conditional formatting syntax, which doesn't make much sense without specifying the false condition...

Reply:
I thought it was an attempt to use a [recursive expression](https://www.boost.org/doc/libs/1_83_0/libs/regex/doc/html/boost_regex/syntax/perl_syntax.html#boost_regex.syntax.perl_syntax.recursive_expressions) but that makes even less sense in a format string.

Reply:
Some archaeology reveals that [I wrote that](https://github.com/art-framework-suite/art/commit/cb727fa5a3ce4416d9b644075fe3b6e5d9feae40) when I was young and stupid (been in the group about 4 months). Since I am no longer young, let us assume that the inexperience of youth produced a harmless piece of nonsense until it is proved otherwise.